python-spec/src/somacore/query/query.py

```diff
@@ -757,14 +757,18 @@ class AxisIndexer:
     def _obs_index(self) -> pd.Index:
```

Comment:
all the appearances of `pd.Index` in this class should probably be replaced with `types.IndexLike`.